src/autoreplace_gui.cpp

```diff
@@ -357,28 +352,21 @@ class ReplaceVehicleWindow : public Window {
 				break;
 			}
 
-			case WID_RV_RAIL_ROAD_TYPE_DROPDOWN: {
+			case WID_RV_RAIL_TYPE_DROPDOWN: {
 				Dimension d = {0, 0};
-				switch (this->window_number) {
-					case VEH_TRAIN:
-						for (RailType rt = RAILTYPE_BEGIN; rt != RAILTYPE_END; rt++) {
-							const RailTypeInfo *rti = GetRailTypeInfo(rt);
-							/* Skip rail type if it has no label */
-							if (rti->label == 0) continue;
```

Comment:
What happened to this check? Is it no longer needed? (Here and with road types below)

Reply:
`_sorted_railtypes`/`_sorted_roadtypes` already excludes types with no label.